torchmultimodal/modules/encoders/clip_text_encoder.py

```diff
@@ -39,42 +47,49 @@ def __init__(
         heads: int = 8,
         layers: int = 12,
         use_clip_init: bool = True,
+        pretrained_url: Optional[str] = None,
     ):
         super().__init__()
-        self.encoder = TransformerEncoder(
-            vocab_size=vocab_size,
-            embedding_dim=width,
-            padding_idx=-1,
-            max_seq_len=context_length,
-            num_encoder_layers=layers,
-            num_attention_heads=heads,
+        self.token_embedding = torch.nn.Embedding(vocab_size, width)
+        self.positional_embedding = torch.nn.Parameter(
+            torch.empty(context_length, width)
+        )
+        encoder_layer = TransformerEncoderLayer(
+            d_model=width,
+            nhead=heads,
             dropout=0.0,
-            normalize_before=True,
+            activation=quick_gelu,
+            norm_first=True,
         )
+        self.encoder = TransformerEncoder(encoder_layer, num_layers=layers)
```

Comment:
this is to utilize PT core's better transformer? I thought text team has or is going to adopt it?

Reply:
They have, but they are also wrapping it with some other stuff (for instance their class also includes the embeddings).

Reply:
This should ultimately be updated by our unified TransformerEncoder

Reply:
the goal here is to get rid of the layer norm after embedding, using core's transformer is just a side effect